gradle/publish.gradle

```diff
@@ -107,11 +107,15 @@ publishing {
   repositories {
     maven {
       // Use specified mavenRepository if provided, else use release or snapshot defaults.
-      url = project.findProperty("mavenRepository") ||
+      url = project.findProperty("mavenRepository") ?:
           project.isReleaseVersion ? project.mavenReleaseUrl : project.mavenSnapshotUrl
-      credentials {
-        username project.findProperty("mavenUsername")
-        password project.findProperty("mavenPassword")
+      if (url.toString().startsWith("http") || url.toString().startsWith("sftp")) {
```

Comment:
@pivotal-jbarrett Yep, you were exactly right.  For preference, the [referenced documentation](https://docs.gradle.org/current/userguide/repository_types.html) specifies credentials should be set for HTTP, HTTPS, and SFTP, so I went with an opt-in rather than a "not-GCS" style here.